test/abel.lean

```diff
@@ -72,3 +72,59 @@ example [AddCommGroup α] (x y z : α) : y = x + z - (x - y + z) := by
 
 -- https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/abel.20bug.3F/near/368707560
 example [AddCommGroup α] (a b s : α) : -b + (s - a) = s - b - a := by abel_nf
+
+/--
+error: abel_nf made no progress
+-/
+#guard_msgs in
+example : False := by abel_nf
+
+/--
+error: abel_nf made no progress
+-/
+#guard_msgs in
+example [AddCommGroup α] (x y z : α) (w : x = y + z) : False := by
+  abel_nf at w
+
+example [AddCommGroup α] (x y z : α) (h : False) (w : x - x = y + z) : False := by
+  abel_nf at w
+  guard_hyp w : 0 = y + z
+  assumption
+
+/--
+error: abel_nf made no progress
+-/
+#guard_msgs in
+example [AddCommGroup α] (x y z : α) (_w : x = y + z) : False := by
+  abel_nf at *
+
+/--
+error: no goals to be solved
+-/
+-- This error message is confusing: it is saying that it closed the main goal,
+-- and so then had nothing to do on the hypotheses.
+-- The user has to guess that they should remove the `at *`.
```

Comment:
I'm guessing here that this error message lies outside the scope of `abel_nf`?

Reply:
Yes, this is the behaviour of `withLocation` in Lean. I've PR'd a better doc-string that at least documnents `withLocaation`'s current behaviour, and will think separately about if it should/could change.